src/plugins/utils.ts

```diff
@@ -0,0 +1,8 @@
+export const insertPluginsToArray = <T>(pluginsArray: any[], array: T[]): T[] => {
+  const components = [...array];
+  pluginsArray.forEach(plugin => {
+    const {priority, ...pluginData} = plugin;
+    priority ? components.splice(priority, 0, pluginData) : components.push(pluginData);
```

Comment:
* Worth to consider unifying `priority`/`order` naming
* `if/else`,
* the comment from above for plugins `order`